supervision/metrics/mean_average_precision.py

```diff
@@ -626,3 +229,1253 @@ def plot(self):
+            self.stats = _summarize_predictions()
+
+    def evaluate(self):
+        """
+        Start the per image evaluation on all images and keeep results in
+        self.eval_imgs (a list of dictionaries).
+        """
+        # Select all parameters to evaluate
+        self.params.img_ids = list(np.unique(self.params.img_ids))
+        self.params.cat_ids = list(np.unique(self.params.cat_ids))
+        self.params.max_dets = sorted(self.params.max_dets)
+
+        self._prepare_targets_and_predictions()
+
+        # Compute IOUs between all targets and predictions for all images and categories
+        self.ious = {
+            (img_id, cat_id): self._compute_iou(img_id, cat_id)
+            for img_id in self.params.img_ids
+            for cat_id in self.params.cat_ids
+        }
+
+        # Select the largest max area (the last element containing 100 dets
+        max_det = self.params.max_dets[-1]
+
+        # Evaluate each image with all categories, area range and max detections
+        self.eval_imgs = [
+            self._evaluate_image(img_id, cat_id, area_range, max_det)
+            for cat_id in self.params.cat_ids
+            for area_range in self.params.area_range
+            for img_id in self.params.img_ids
+        ]
+
+        # Accumulate results
+        self._accumulate()
+
+
+class MeanAveragePrecision(Metric):
+    """
+    Mean Average Precision (mAP) is a metric used to evaluate object detection models.
+    It is the average of the precision-recall curves at different IoU thresholds.
+
+    Example:
+        ```python
+        import supervision as sv
+        from supervision.metrics import MeanAveragePrecision
+
+        predictions = sv.Detections(...)
+        targets = sv.Detections(...)
+
+        map_metric = MeanAveragePrecision()
+        map_result = map_metric.update(predictions, targets).compute()
+
+        print(map_result.map50_95)
+        # 0.4674
+
+        print(map_result)
+        # MeanAveragePrecisionResult:
+        # Metric target: MetricTarget.BOXES
+        # Class agnostic: False
+        # mAP @ 50:95: 0.4674
+        # mAP @ 50:    0.5048
+        # mAP @ 75:    0.4796
+        # mAP scores: [0.50485  0.50377  0.50377  ...]
+        # IoU thresh: [0.5  0.55  0.6  ...]
+        # AP per class:
+        # 0: [0.67699  0.67699  0.67699  ...]
+        # ...
+        # Small objects: ...
+        # Medium objects: ...
+        # Large objects: ...
+
+        map_result.plot()
+        ```
+
+    ![example_plot](\
+        https://media.roboflow.com/supervision-docs/metrics/mAP_plot_example.png\
+        ){ align=center width="800" }
+    """
+
+    def __init__(
+        self,
+        metric_target: MetricTarget = MetricTarget.BOXES,
+        class_agnostic: bool = False,
+        class_mapping: Optional[Dict[int, int]] = None,
+        image_indices: Optional[List[int]] = None,
+    ):
+        """
+        Initialize the Mean Average Precision metric.
+
+        Args:
```

Comment:
Update docstring to reflect newly added arguments.

Reply:
My bad. Thanks for noticing it :smile: